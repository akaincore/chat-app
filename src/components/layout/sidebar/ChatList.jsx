import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ChatListItem from './ChatListItem';
import Chat from '../../../types/Chat';

const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
    overflowY: 'auto',
  },
  chatList: {
    paddingBottom: '60px',
  },
});

const ChatList = ({
 classes, chats, activeChat, disabled,
}) => (
  <div className={classes.root}>
    <List className={classes.chatList}>
      {chats.map(chat => (
        <ChatListItem
          disabled={disabled}
          key={chat._id}
          chatId={chat._id}
          selected={activeChat && activeChat._id === chat._id}
          {...chat}
        />
      ))}
    </List>
  </div>
);

ChatList.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  chats: PropTypes.arrayOf(PropTypes.shape(Chat)).isRequired,
  activeChat: PropTypes.shape(Chat),
  disabled: PropTypes.bool.isRequired,
};

ChatList.defaultProps = {
  activeChat: null,
};

export default withStyles(styles)(ChatList);
