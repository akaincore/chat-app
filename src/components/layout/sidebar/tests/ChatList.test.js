/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatList from '../ChatList';

jest.mock('../ChatListItem', () => () => 'ChatListItem');

const mockProps = {
  chats: [],
  disabled: false,
  activeChat: {
    _id: '12345',
    creator: {
      username: 'user',
    },
    title: 'foo',
  },
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatList {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
