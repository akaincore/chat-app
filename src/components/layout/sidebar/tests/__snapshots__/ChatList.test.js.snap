// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `
<div
  className="ChatList-root-7"
>
  <ul
    className="MuiList-root-9 MuiList-padding-10 ChatList-chatList-8"
  />
</div>
`;
